File: sample/python/main_change_material_sample.py
from time import sleep
# voxelammingパッケージからVoxelammingクラスをインポートします
from voxelamming import Voxelamming
# from voxelamming_local import Voxelamming  # ローカルで開発している場合はこちらを使う

# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# Voxelammingクラスのインスタンスを生成します
vox = Voxelamming(room_name)
# ボクセルの設定を行います
vox.set_box_size(1)
vox.set_build_interval(0.01)

# ボクセルを配置するため、位置と色を設定します
colors = [
    [0, 0, 0],
    [1, 0, 0],
    [0, 1, 0],
    [0, 0, 1],
    [1, 1, 0],
    [1, 0, 1],
    [0, 1, 1],
    [1, 1, 1],
    [0.5, 0.5, 0.5],
    [0.5, 0, 0],
    [0, 0.5, 0],
    [0, 0, 0.5],
    [0.5, 0.5, 0],
    [0.5, 0, 0.5],
]

for i, color in enumerate(colors):
    vox.create_box(0, i, 0, *color, alpha=1)

for i in range(5):
    vox.change_material(is_metallic=False, roughness=0.25 * i)
    vox.transform(i, 0, 0, pitch=0, yaw=0, roll=0)
    # ボクセルデータをアプリに送信します。
    vox.send_data()
    sleep(0.1)

for i in range(5):
    vox.change_material(is_metallic=True, roughness=0.25 * i)
    vox.transform(5 + i, 0, 0, pitch=0, yaw=0, roll=0)
    # ボクセルデータをアプリに送信します。
    vox.send_data()
    sleep(0.1)

File: sample/javaScript/mainChangeMaterialSample.mjs
import { Voxelamming } from 'voxelamming';
// import Voxelamming from './voxelamming.js';  // test

let roomName = "1000";
let vox = new Voxelamming(roomName);

vox.setBoxSize(1);
vox.setBuildInterval(0.01);

let colors = [
  [0, 0, 0],
  [1, 0, 0],
  [0, 1, 0],
  [0, 0, 1],
  [1, 1, 0],
  [1, 0, 1],
  [0, 1, 1],
  [1, 1, 1],
  [0.5, 0.5, 0.5],
  [0.5, 0, 0],
  [0, 0.5, 0],
  [0, 0, 0.5],
  [0.5, 0.5, 0],
  [0.5, 0, 0.5],
  [0, 0.5, 0.5],
];

for (let i = 0; i < colors.length; i++) {
  let color = colors[i];
  vox.createBox(0, i, 0, color[0], color[1], color[2], 1);
}

for (let i = 0; i < 5; i++) {
  vox.changeMaterial(false, 0.25 * i);
  vox.transform(i, 0, 0,  0, 0, 0);
  await vox.sendData();
  await  vox.sleepSecond(0.1)
}

for (let i = 0; i < 5; i++) {
  vox.changeMaterial(true, 0.25 * i);
  vox.transform(5 + i, 0, 0, 0, 0, 0);
  await vox.sendData();
  await  vox.sleepSecond(0.1)
}

